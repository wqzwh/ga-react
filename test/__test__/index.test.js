import React, { Component } from 'react'
import { render, mount } from 'enzyme'
import { Button } from 'antd'
describe('Button', () => {
  it('按钮测试', () => {
    const wrapper = render(<Button>测试</Button>)
    expect(wrapper).toMatchSnapshot()
  })

  it('测试加载中', () => {
    class DefaultButton extends Component {
      constructor(props) {
        super(props)
        this.state = {
          loading: false
        }
        this.enterLoading = this.enterLoading.bind(this)
      }
      enterLoading() {
        this.setState({ loading: true })
      }
      render() {
        return (
          <Button loading={this.state.loading} onClick={this.enterLoading}>
            Button
          </Button>
        )
      }
    }
    const wrapper = mount(<DefaultButton />)
    wrapper.simulate('click')
    expect(wrapper.find('.ant-btn-loading').length).toBe(1)
  })
})
