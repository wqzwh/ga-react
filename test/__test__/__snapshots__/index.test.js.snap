// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button 按钮测试 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "ant-btn",
      "type": "button",
    },
    "children": Array [
      Object {
        "attribs": Object {},
        "children": Array [
          Object {
            "data": "测 试",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "span",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
    ],
    "name": "button",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "type": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "type": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
